mypy/stubgen.py

```diff
@@ -1010,6 +1012,37 @@ def get_base_types(self, cdef: ClassDef) -> list[str]:
             elif isinstance(base, IndexExpr):
                 p = AliasPrinter(self)
                 base_types.append(base.accept(p))
+            elif isinstance(base, CallExpr):
+                # namedtuple(typename, fields), NamedTuple(typename, fields) calls can
+                # be used as a base class. The first argument is a string literal that
+                # is usually the same as the class name.
+                #
+                # Note:
+                # A call-based named tuple as a base class cannot be safely converted to
+                # a class-based NamedTuple definition because class attributes defined
+                # in the body of the class inheriting from the named tuple call are not
+                # namedtuple fields at runtime.
+                if self.is_namedtuple(base):
+                    nt_fields = self._get_namedtuple_fields(base)
+                    assert isinstance(base.args[0], StrExpr)
+                    typename = base.args[0].value
+                    if nt_fields is not None:
+                        # A valid namedtuple() call, use NamedTuple() instead with
+                        # Incomplete as field types
+                        fields_str = ", ".join(f"({f!r}, {t})" for f, t in nt_fields)
+                    else:
+                        # Invalid namedtuple() call, cannot determine fields
```

Comment:
Looks like you generate `NamedTuple(name, [Incomplete])` as the base in this case. That seems wrong, as it's an invalid type. I think we should make `Incomplete` the base in this case.

Reply:
Also, is this code path tested?

Reply:
Good call. I fixed it. The tests now cover the invalid namedtuple call code path.